### DISCUSSION TITLE
Why doesn't the porosity_profile metric return normalized data (max 1.0)?

### DISCUSSION THREAD

**Discussion by jmcmt87** [PMEAL/porespy#583] [Q&A] [answered]
The results I should get should be between 0 and 1, but I get three arrays with numbers always within the range of 105 and 120, why is that? How can I fix it? 

Thank you in advance

**Comment by jgostick** [comment, marked as answer]
I'm guessing your image is not binary?  Like has 0's and 255's for instance?  Can you check this?  If I'm correct, then I guess we should alter the function to only accept binary images.

**Reply by jmcmt87** [reply]
It doesn't have 0s or 255s all values are between 30 and 180. It's a grey image so I guess it's a binary one

**Reply by ma-sadeghi** [reply]
@jmcmt87 Can you try mapping it to a boolean image, i.e. 30->0, 180->1 and try again?

**Reply by jmcmt87** [reply]
Sorry, I just double checked it and there are just a few 0s and 255s


Also, I tried dividing the ps.metrics.porosity_profile(im, n) by 255 and all results in the 3 dimensions get placed in a very narrow range (0.44 and 0.46) I didn't expect this but not sure if it's normal or if it made any sense for me to divided it by 255

**Reply by ma-sadeghi** [reply]
Ok, so if there are a few 0s and 255s, you need to somehow make your image binary (i.e., consisting of only two labels). Then, like I said before, you need to map your image to 0s and 1s. Let's say your image consists of 20s and 35s, then you do:
```python
im[im==20] = 0
im[im==30] = 1
```
Then pass your image to `porosity_profile`. 

(I haven't actually used this method, but from the looks of it, this is how it should work)

**Reply by jgostick** [reply]
Now that I think of it, I believe this is why we don't automatically convert the image to boolean, since the user must make that conversion in the case of greyscale images.  It sounds like @jmcmt87 has an actual grey scale image that needs thresholding into binary (0 = solid, 1 = void).  The other thing that I recall is that we devised the funciton to work on "ternary" images or higher, so that a user could put 2's in regions that should be ignored.
